### ISSUE TITLE
pod crashes running apt update

### ISSUE THREAD

**Issue by TitaniumCoder477** [containers/podman#22211]
### Issue Description

I noticed this issue just today when I was attempting to update update packages within some pods. The symptom affects my Debian and Ubuntu pods, not my Fedora pods. I can run apt update just fine, but when I run apt upgrade -y, within a few seconds, right in the middle of the update of packages, the pod crashes and returns me to the host. If I run podman logs, the only error is: "Error: An error occurred".

Any other running pods remain running. Only the pod I was updating packages in at that moment crashes. I found this symptom afflicting my existing pods and new pods. I created a pod based on Debian 11 (all my existing pods are 12), and it also experienced the symptom.

Figure 1: test1 pod is running and preparing to install inkscape packages

![2024-03-28_20-26](https://github.com/containers/podman/assets/26101661/fdb189f8-5b62-4c0b-8c5e-985db1053962)

Figure 2: test1 pod is no longer running (crashed)
![2024-03-28_20-26_1](https://github.com/containers/podman/assets/26101661/f8d95230-48a3-42b8-9a4e-daea5b2a1435)

Figure 3: Error message
![2024-03-28_20-26_2](https://github.com/containers/podman/assets/26101661/5ee57e6f-917e-444a-b5f8-ea28a40a8d9f)

### Steps to reproduce the issue

Steps to reproduce the issue
1. Create a brand new pod (I use distrobox, but the symptom happens even when updating packages through podman desktop)
_distrobox create --name test1 --image debian:11 --home /home/jwilmoth/distroboxhomes/test1_
2. Enter your pod
_distrobox enter test1   (or podman desktop > terminal)_
3. Wait for it to initialize
4. sudo apt update && sudo apt upgrade -y
_If there are no updates, then go ahead and install something, anything. If the error isn't exhibited, try again. I have found the symptom occuring on demand during an install or update of anywhere between 5 and 10 packages._

### Describe the results you received

The pod crashes in the middle of the update, and after starting it up again and connecting, I have to run dpkg --configure -a before I can retry. Often it crashes again on the retry, depending on how much needed to be done.

### Describe the results you expected

I expect the pod to remain running like it always has even during an installation or update of one or more packages.

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.33.5
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.10-1.fc39.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.10, commit: '
  cpuUtilization:
    idlePercent: 98.86
    systemPercent: 0.29
    userPercent: 0.85
  cpus: 40
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: kde
    version: "39"
  eventLogger: journald
  freeLocks: 1922
  hostname: gideon
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 524288
      size: 65536
  kernel: 6.7.10-200.fc39.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 185494237184
  memTotal: 202728189952
  networkBackend: netavark
  networkBackendInfo:
    backend: netavark
    dns:
      package: aardvark-dns-1.10.0-1.fc39.x86_64
      path: /usr/libexec/podman/aardvark-dns
      version: aardvark-dns 1.10.0
    package: netavark-1.10.3-1.fc39.x86_64
    path: /usr/libexec/podman/netavark
    version: netavark 1.10.3
  ociRuntime:
    name: crun
    package: crun-1.14.4-1.fc39.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.14.4
      commit: a220ca661ce078f2c37b38c92e66cf66c012d9c1
      rundir: /run/user/1000/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  pasta:
    executable: /usr/bin/pasta
    package: passt-0^20240220.g1e6f92b-1.fc39.x86_64
    version: |
      pasta 0^20240220.g1e6f92b-1.fc39.x86_64-pasta
      Copyright Red Hat
      GNU General Public License, version 2 or later
        <https://www.gnu.org/licenses/old-licenses/gpl-2.0.html>
      This is free software: you are free to change and redistribute it.
      There is NO WARRANTY, to the extent permitted by law.
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.2-1.fc39.x86_64
    version: |-
      slirp4netns version 1.2.2
      commit: 0ee2d87523e906518d34a6b423271e4826f71faf
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 15462293504
  swapTotal: 15462293504
  uptime: 0h 37m 29.00s
  variant: ""
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/jwilmoth/.config/containers/storage.conf
  containerStore:
    number: 23
    paused: 0
    running: 1
    stopped: 22
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/jwilmoth/.local/share/containers/storage
  graphRootAllocated: 992793853952
  graphRootUsed: 653326581760
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Supports shifting: "false"
    Supports volatile: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 8
  runRoot: /run/user/1000/containers
  transientStore: false
  volumePath: /home/jwilmoth/.local/share/containers/storage/volumes
version:
  APIVersion: 4.9.3
  Built: 1708357294
  BuiltTime: Mon Feb 19 10:41:34 2024
  GitCommit: ""
  GoVersion: go1.21.7
  Os: linux
  OsArch: linux/amd64
  Version: 4.9.3
```

### Podman in a container

No

### Privileged Or Rootless

None

### Upstream Latest Release

No

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by baude** [comment]
do you have a reproducer using only podman?  or just podman restful api? otherwise, i am wondering if this better suited to be filed against podman-desktop.

**Comment by TitaniumCoder477** [comment]
I'm not sure what you're asking, but it sounds like eliminating podman-desktop. I uninstalled it just now and re-tested. The symptom remains.

Figure 1: distrobox enter test1, followed by sudo apt install openttd
![2024-03-29_17-40](https://github.com/containers/podman/assets/26101661/ad3ff2ac-6f3d-4d96-94ef-3d1af37f7c9f)

Figure 2: once pressing 'y', installation starts and barely gets off the ground before container crashes and returns to host
![2024-03-29_17-40_1](https://github.com/containers/podman/assets/26101661/3b3ed64e-d534-45c1-b55a-6504b56cb46e)

Figure 3: podman logs test1 shows same error message
![2024-03-29_17-41](https://github.com/containers/podman/assets/26101661/571effb6-8d98-4217-81d9-9817dd098d0d)

**Comment by baude** [comment]
im not asking you to uninstall podman-desktop, just take it out of the equation.  using only the podman cli, can you give us a reproducer for your problem ?

**Comment by TitaniumCoder477** [comment]
Are we speaking the same language? I said "eliminating podman-desktop" and you said "take it out of the equation." They are the same thing practically-speaking. I only installed podman-desktop recently and don't really use it, so I had no issues doing that. Like I said, I use distrobox, not podman directly and not podman-desktop.

I just re-tested again now, and I am not getting the symptom anymore. I tested updated three containers, even simultaneously, without any issues whatsoever. Super odd. I have no idea why this symptom would have appeared and then disappeared, without any rhyme or reason. Any closing thoughts before we close out this issue?
